### PR TITLE
Upgrade app octane

### LINKED ISSUE #603: Upgrade App to Octane

### PR DESCRIPTION
**Pull request by kevinhinterlong**
Resolves #603
Addresses app for #617 

## Description
Upgrade app to octane

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/app/.eslintrc.js [hunk 1]
```diff
@@ -11,11 +11,12 @@ module.exports = {
...
   extends: ['eslint:recommended', 'plugin:ember/recommended', 'prettier'],
   env: {
-    browser: true
+    browser: true,
+    es6: true
   },
   rules: {
     'multiline-comment-style': ['error', 'starred-block'],
-    'ember/no-jquery': 'error'
+    'ember/no-jquery': 'off'
```
Comment (jkusa) [inline packages/app/.eslintrc.js]: Do we use jQuery in the app or only in tests?

Comment (kevinhinterlong) [reply packages/app/.eslintrc.js]: We use it in one test

File: packages/app/app/routes/application.js [hunk 2]
```diff
@@ -1,33 +1,28 @@
...
-  model() {
-    return hash({
-      user: this.user.findOrRegister(),
-      metadata: this.bardMetadata.loadMetadata()
-    }).then(() => {
-      return;
-    });
+  async model() {
+    await this.user.findOrRegister();
+    await this.bardMetadata.loadMetadata();
```
Comment (jkusa) [inline packages/app/app/routes/application.js]: Can we do:
```js
await Promse.all([
  this.user.findOrRegister(),
  this.bardMetadata.loadMetadata();
]);
```

File: packages/reports/addon/templates/reports/report/edit.hbs [hunk 3]
```diff
@@ -8,7 +8,7 @@
   {{#with (model-for (parent-route)) as | report |}}
     <p>
       Select some dimensions and metrics, then click
-      <a class="navi-link clickable"
+      <a href class="navi-link clickable"
```
Comment (jkusa) [inline packages/reports/addon/templates/reports/report/edit.hbs]: `{{!-- template-lint-disable link-href-attributes --}}`?

We should eventually convert all these non `href` `<a>` elements to something else.

File: packages/app/.eslintrc.js (hunk 1 above)
Comment (jkusa) [reply packages/app/.eslintrc.js]: If so, can we turn this off only in the test dir?

File: packages/reports/addon/templates/reports/report/edit.hbs (hunk 3 above)
Comment (kevinhinterlong) [reply packages/reports/addon/templates/reports/report/edit.hbs]: Good catch, idk how I messed this one up. Yea I'll make an issue for this so that it we can track it

File: packages/app/.eslintrc.js [hunk 4]
```diff
@@ -3,12 +3,16 @@ module.exports = {
...
   },
   plugins: ['ember'],
   extends: ['eslint:recommended', 'plugin:ember/recommended', 'prettier'],
   env: {
-    browser: true
+    browser: true,
+    es6: true
   },
   rules: {
     'multiline-comment-style': ['error', 'starred-block']
```
Comment (kevinhinterlong) [inline packages/app/.eslintrc.js]:
```suggestion
    'multiline-comment-style': ['error', 'starred-block'],
    'ember/no-jquery': 'error'
```
Add back root level no jquery rule

File: packages/core/addon/templates/components/navi-visualization-config/line-chart.hbs [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
-  "navi-visualization-config/series-chart"
-  request=request
-  response=response
-  seriesConfig=seriesConfig
-  seriesType=seriesType
-  onUpdateConfig=(action "onUpdateConfig")
-}}
+{{!-- Copyright 2020, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
+<NaviVisualizationConfig::SeriesChart
+  @request={{request}}
```
Comment (mattgraham1995) [inline packages/core/addon/templates/components/navi-visualization-config/line-chart.hbs]: should these be `@request={{@request}}`? I know it's in core so it's not on octane yet, but it should still work I think

File: packages/dashboards/addon/templates/dashboards/dashboard/widgets/widget.hbs [hunk 6]
```diff
@@ -119,9 +119,17 @@
   {{!-- Footer --}}
 
   <footer class="navi-report-widget__footer">
-    {{#if (not isRunningReport) class="navi-report-widget__footer-btns"}}
-
+    {{#if isRunningReport}}
+      <button
```
Comment (mattgraham1995) [inline packages/dashboards/addon/templates/dashboards/dashboard/widgets/widget.hbs]: Is this addition necessary for this PR? It might be best to add `cancelReport` to the widget view in a separate PR and add a test around it just so we know it works in that view as well as the regular report view.

File: packages/reports/addon/templates/components/common-actions/get-api.hbs [hunk 7]
```diff
@@ -47,7 +47,7 @@
         Copy Link
       {{/copy-button}}
       <a class="btn btn-secondary" href={{requestUrl}} target="_blank" rel="noopener noreferrer">Run API Query in New Tab</a>
-      <button class="btn btn-secondary" onclick={{pipe (toggle "showModal" this) (action "resetNotifications")}}>Cancel</button>
+      <button type="button" class="btn btn-secondary" onclick={{pipe (toggle "showModal" this) (action "resetNotifications")}}>Cancel</button>
```
Comment (mattgraham1995) [inline packages/reports/addon/templates/components/common-actions/get-api.hbs]:
```suggestion
      <button type="button" class="btn btn-secondary" onclick={{queue (toggle "showModal" this) (action "resetNotifications")}}>Cancel</button>
```

File: packages/reports/addon/templates/components/common-actions/share.hbs [hunk 8]
```diff
@@ -45,7 +45,7 @@
       }}
         Copy Link
       {{/copy-button}}
-      <button class="btn btn-secondary" onclick={{pipe (toggle "showModal" this) (action "resetNotifications")}}>Cancel</button>
+      <button type="button" class="btn btn-secondary" onclick={{pipe (toggle "showModal" this) (action "resetNotifications")}}>Cancel</button>
```
Comment (mattgraham1995) [inline packages/reports/addon/templates/components/common-actions/share.hbs]:
```suggestion
      <button type="button" class="btn btn-secondary" onclick={{queue (toggle "showModal" this) (action "resetNotifications")}}>Cancel</button>
```

File: packages/core/addon/templates/components/navi-visualization-config/line-chart.hbs (hunk 5 above)
Comment (kevinhinterlong) [reply packages/core/addon/templates/components/navi-visualization-config/line-chart.hbs]: Yea it should work, I was making minimal changes in other packages to satisfy template lint
